src/pages/myAppointments.tsx

```diff
@@ -0,0 +1,58 @@
+import AppointmentCard from "@/components/appointment/AppointmentCard";
+import Navbar from "@/components/public/Navbar";
+import { GetAppointmentsByUserId } from "@/services/Appointment";
+import { Appointment } from "@/types/Appointment";
+import { Typography, Grid, Box } from "@mui/material";
+import { useSupabaseClient } from "@supabase/auth-helpers-react";
+import { useContext, useEffect, useState } from "react";
+import { Database } from "supabase/db_types";
+import { userContext } from "supabase/user_context";
+import { NextRouter, useRouter } from "next/router";
+import { PagePaths } from "enum/pages";
+
+export default function Home() {
+    const router: NextRouter = useRouter();
+
+    const supabaseClient = useSupabaseClient<Database>();
+    const [appointments, setAppointments] = useState<Appointment[]>([]);
+    const userStatus = useContext(userContext);
+
+    function handleCardClick(appointmentId: string): void {
+        router.push(PagePaths.myAppointments + appointmentId);
+    }
+
+    useEffect(() => {
+        if (!userStatus.user) return;
+
+        GetAppointmentsByUserId(userStatus.user.userId, supabaseClient).then((appointment) => {
+            setAppointments(appointment);
+        }).catch((err) => {
+            console.log(err)
+            return;
+        })
+
+    }, [supabaseClient, userStatus.user]);
+
```

Comment:
Need guard to check whether the user is logged in using userStatus above. Loading UI should also be here since it take some time to get data from backend. Add a new state isLoading if it is still true return <Loading /> instead of the normal UI. New state have to be used because checking if appointments is empty list or not could not distinguish between whether the user has no appointment or the process of getting data hasn't completed yet.

Reply:
done fixed it